packages/generator/src/generator.ts

```diff
@@ -85,41 +85,28 @@ export async function generateWithParsedOptions(
     setLogLevel('verbose', logger);
   }
 
-  const projectAndServices = await generateProject(
-    options as ParsedGeneratorOptions
-  );
+  const projectAndServices = await generateProject(options);
   if (!projectAndServices) {
     throw Error('The project is undefined.');
   }
-  const services = projectAndServices.services;
+  const { services } = projectAndServices;
 
-  await generateFilesWithoutTsMorph(
-    services,
-    options as ParsedGeneratorOptions
-  );
+  await generateFilesWithoutTsMorph(services, options);
 
   if (options.transpile) {
     const directories = services
       .filter(async service => {
         const files = await readdir(
-          resolvePath(service.directoryName, options as ParsedGeneratorOptions)
+          resolveOutDirPath(service.directoryName, options)
```

Comment:
[pp] As I understood are at this point the paths already resolved. I would therefore use `join` and name the method accordingly.

Reply:
I now deleted the old function and everywhere it was called I am using `join(outputDir, path)`